nixos/modules/services/monitoring/collectd.nix

```diff
@@ -100,27 +85,21 @@ in {
       wantedBy = [ "multi-user.target" ];
 
       serviceConfig = {
-        ExecStart = "${cfg.package}/sbin/collectd -C ${conf} -P ${cfg.pidFile}";
-        Type = "forking";
-        PIDFile = cfg.pidFile;
-        User = optional (cfg.user!="root") cfg.user;
+        ExecStart = "${cfg.package}/sbin/collectd -C ${conf} -f";
+        User = cfg.user;
+        AmbientCapabilities = "cap_net_raw cap_dac_override"; # cap_net_raw for ping, cap_dac_override for smart
```

Comment:
I am a bit concerned about enabling `cap_dac_override` by default, because we would give a network reachable daemon written in C the permission to override every file on the system. Also looking at the upstream service, it seems that for the smart plugin they refer to `CAP_SYS_RAWIO`: https://github.com/collectd/collectd/blob/master/contrib/systemd.collectd.service. In this file they also do not set any capabilities by default but leave it up to the user, what is required. We could do the same by pointing to the upstream documentation and add a note like:

Some plugins may require more [capabilities](https://github.com/collectd/collectd/blob/master/contrib/systemd.collectd.service#L18). To enable them set:

```
systemd.services.collectd.serviceConfig.AmbientCapabilties = ["cap_net_raw", "cap_dac_override"];
```

(this could be also a dedicated option of collectd of course).